frontend/server/src/Controllers/Problem.php

```diff
@@ -4408,33 +4420,30 @@ public static function getProblemEditDetailsForSmarty(
                 return $result;
             }
         } elseif ($r['request'] === 'markdown') {
-            [
-                'problem' => $problem,
-            ] = self::validateCreateOrUpdate($r->identity, $problemParams);
-            if (is_null($problem)) {
-                throw new \OmegaUp\Exceptions\NotFoundException(
-                    'problemNotFound'
-                );
-            }
             \OmegaUp\Validators::validateStringNonEmpty(
-                $r['wmd-input-statement'],
-                'statement'
+                $r['directory'],
+                'directory'
             );
-            \OmegaUp\Validators::validateOptionalInEnum(
-                $r['statement-language'],
-                'lang',
-                \OmegaUp\Controllers\Problem::ISO639_1
-            );
-
-            self::updateStatement(
-                $r->identity,
-                $r->user,
-                $problem,
-                $r['wmd-input-statement'],
-                $r['message'],
-                $r['statement-language'],
-                $problemParams->updatePublished
+            \OmegaUp\Validators::validateStringNonEmpty(
+                $r['wmd-input'],
+                'wmd-input'
             );
+            /** @var array<string, string> $statements */
+            $statements = json_decode($r['wmd-input'], true);
```

Comment:
se puede validar que efectivamente sea un arreglo que contenga cadenas? onda esto debería anotarse como `mixed`. también

```suggestion
            $statements = json_decode($r['wmd-input'], /*$assoc=*/true);
```

Reply:
Listo